marketdata/stream/client.go

```
@@ -216,6 +221,24 @@ func (cc *CryptoClient) constructURL() (url.URL, error) {
 	return constructURL(cc.baseURL, cc.feed)
 }
 
+// This will create a crytpo perpetual futures subscriptions to the Alpaca market data services
+type CryptoPerpsClient struct {
+	CryptoClient
+}
+
+// NewCryptoPerpsClient returns a new CryptoPerpsClient that will connect to the crypto perpetual futures feed
+// Currently only the marketdata.GLOBAL is supported.
+// Base URL will be modified to reflect the perpetual endpoint with respect to the crypto endpoint
+func NewCryptoPerpsClient(feed marketdata.CryptoFeed, opts ...CryptoOption) *CryptoPerpsClient {
+	if feed != marketdata.GLOBAL {
+		return nil
+	}
```

Don't do this, it will at best cause a panic and at worst a silent error.